src/main/java/array/Array.java

```diff
@@ -0,0 +1,49 @@
+package array;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * Interface for an array with undefined number of dimensions
+ *
+ * @param <T> - type of values that allow to put into array
+ * @param <C> - coordinates class for navigation through array
+ */
+public interface Array<T, C> {
+
+    /**
+     * Insert value in array
+     *
+     * @param coordinates - index of inserting
+     * @param value - value to insert
+     */
+    void insert(C coordinates, T value);
```

Comment:
don't you want to return inserted value here?

Reply:
in case of the course - not, but your implementation is so mature-looking that I thought it'll be good to do that )

Reply:
Fixed at 9ab394d2d378fc1a0c5f9a0dfc5879a7e2c60c42